common/desktop-exports

```diff
@@ -391,13 +391,18 @@ fi
 # Icon themes cache
 if [ $needs_update = true ]; then
   rm -rf $XDG_DATA_HOME/icons
-  mkdir -p $XDG_DATA_HOME/icons
+  ensure_dir_exists $XDG_DATA_HOME/icons
   for ((i = 0; i < ${#data_dirs_array[@]}; i++)); do
+    # The runtime and theme content snaps should already contain icon caches
+￼   # so we skip them to optimise app start time.
+￼   if [[ "${data_dirs_array[$i]}" == "$SNAP/data-dir" || "${data_dirs_array[$i]}" == "$RUNTIME/"* ]]; then
+￼     continue
+￼   fi
```

Comment:
All looks great! My only quibble would be that lines 397 to 400 are all one space short of the correct indentation :P